plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -228,6 +234,7 @@
     private static final Set<String> GLACIER_STORAGE_CLASSES = ImmutableSet.of(Glacier.toString(), DeepArchive.toString());
     private static final MediaType DIRECTORY_MEDIA_TYPE = MediaType.create("application", "x-directory");
     private static final String S3_DEFAULT_ROLE_SESSION_NAME = "trino-session";
+    private static final int BATCH_DELETE_OBJECTS_REQEST_MAX_PAGE_SIZE = 1000;
```

Comment:
Non-blocking: Should this be configurable?

Reply:
> when would a user want to set it?

I think there is no need in being able to customize this setting.

As can be seen from aws-sdk-java `DeleteObjectsRequest` javadoc, there can be specified up to `1000` keys:

https://github.com/aws/aws-sdk-java/blob/8f6c93e49024049d2e5d882a3e2cd0e99e55ee65/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/DeleteObjectsRequest.java#L24-L32